document/array.go

```diff
@@ -8,7 +8,7 @@ import (
 // ErrValueNotFound must be returned by Array implementations, when calling the GetByIndex method and
 // the index wasn't found in the array.
 var (
-	ErrValueNotFound   = errors.New("value not found")
```

Comment:
Please avoid changes that are not related to the PR, this can make rebases more difficult for everyone